packages/experiments/src/components/FloatingSuggestions/FloatingSuggestions.tsx

```diff
@@ -0,0 +1,352 @@
+          ev.preventDefault();
+          ev.stopPropagation();
+        } else {
+          // no selection. Try to force resolve the current query
+          this._validateAndInsertCurrentQueryString();
+        }
+        break;
+
+      case KeyCodes.del:
+        if (
+          this.props.onRemoveSuggestion &&
+          this.suggestionsControl.current &&
+          this.suggestionsControl.current.hasSuggestionSelected &&
+          this.suggestionsControl.current.currentSuggestion &&
+          ev.shiftKey
+        ) {
+          this.props.onRemoveSuggestion(this.suggestionsControl.current.currentSuggestion!.item);
+
+          this.suggestionsControl.current.removeSuggestion();
+          this.forceUpdate();
+          ev.stopPropagation();
+        }
+        break;
+
+      case KeyCodes.up:
+        if (this.suggestionsControl.current && this.suggestionsControl.current.handleKeyDown(keyCode)) {
+          ev.preventDefault();
+          ev.stopPropagation();
+          this._updateActiveDescendant();
+        }
+        break;
+
+      case KeyCodes.down:
+        if (this.suggestionsControl.current && this.suggestionsControl.current.handleKeyDown(keyCode)) {
+          ev.preventDefault();
+          ev.stopPropagation();
+          this._updateActiveDescendant();
+        }
+        break;
+    }
+  };
+
+  private _updateActiveDescendant(): void {
+    if (this.props.inputElement && this.suggestionsControl.current && this.suggestionsControl.current.selectedElement) {
+      const selectedElId = this.suggestionsControl.current.selectedElement.getAttribute('id');
+      if (selectedElId) {
+        this.props.inputElement.setAttribute('aria-activedescendant', selectedElId);
+      }
+    }
+  }
+
+  private _onResolveSuggestions(updatedValue: string): void {
+    const suggestions: TItem[] | PromiseLike<TItem[]> | null = this.props.onResolveSuggestions(updatedValue, this.props.selectedItems);
+
+    this._updateSuggestionsVisible(true /*shouldShow*/);
+    if (suggestions !== null) {
+      this.updateSuggestionsList(suggestions);
+    }
+  }
+
+  private _validateAndInsertCurrentQueryString = (): void => {
+    if (this.state.queryString && this.props.isQueryForceResolveable && this.props.createForceResolvedItem) {
+      const isForceResolvable: boolean = this.props.isQueryForceResolveable(this.state.queryString);
+      if (!isForceResolvable) {
+        return;
+      }
+
+      const itemToConvert: ISuggestionModel<TItem> = this.props.createForceResolvedItem(this.state.queryString);
+      const convertedItems = this.suggestionStore.convertSuggestionsToSuggestionItems([itemToConvert]);
+      this.onChange(convertedItems[0].item);
+    }
+  };
+
+  private _updateSuggestionsVisible(shouldShow: boolean): void {
+    if (shouldShow) {
+      this.showPicker();
+    } else {
+      this.hidePicker();
+    }
+  }
+
+  private _bindToInputElement(): void {
+    if (this.props.inputElement && !this.state.didBind) {
```

Comment:
I know that we already had this implemented this way, but is there any other option to avoid doing a bind like this? Like the render could wrap an input element as a child and have `onKeyDown` on the surrounding div.

Reply:
Not as things are currently structured. The floatingSuggestions doesn't render the input element. What you're proposing sounds like it would be best solved with the context-based suggestions we talked about, and that I'm planning on taking a stab at after I can get the UnifiedPickers as they are merged in